back/app/models/app_configuration.rb

```diff
@@ -109,7 +109,7 @@ def has_feature?(f)
   def closest_locale_to(locale)
     locale = locale.to_s
     locales = settings.dig('core', 'locales') || []
-    locales.include?(locale) ? locale : locales.first
+    locales.any? { |l| l.include?(locale) } ? locales.find { |l| l.include?(locale) } : locales.first
```

Comment:
I'm wondering whether there is any opportunity for getting it wrong here, if the main language passed in `locale` would somehow wrongly match with a substring because it's contained by chance.

Fictional example: if the locale passed is `af` and some language is called `pt-af` or `paf-BT`. I don't know the language standard enough to exclude those cases, but I think they are excluded because of the uppercase regional part and always being 2 characters? Probably in any case good enough really, and not worth spending more time on.

Reply:
I had much the same train of thought. It's theoretically possible, but probably unlikely / never happens.